tests/multi_ptr/multi_ptr_common.h

```diff
@@ -33,10 +33,48 @@ struct StringMaker<sycl::access::address_space> {
     }
   }
 };
+
+template <>
+struct StringMaker<sycl::access::decorated> {
+  using type = sycl::access::decorated;
+  static std::string convert(type value) {
+    switch (value) {
+      case type::yes:
+        return "access::decorated::yes";
+      case type::no:
+        return "access::decorated::no";
+      case type::legacy:
+        return "access::decorated::legacy";
+      default:
+        return "unknown";
+    }
+  }
+};
 }  // namespace Catch
 
 namespace multi_ptr_common {
 
+/**
+ * @brief Factory function for getting type_pack with fields of
+ *        sycl::access::address_space enums
+ */
+inline auto get_address_spaces() {
+  return value_pack<
+      sycl::access::address_space, sycl::access::address_space::global_space,
+      sycl::access::address_space::local_space,
+      sycl::access::address_space::private_space,
+      sycl::access::address_space::generic_space>::generate_named();
```

Comment:
We might need to modify the test later for implementations that don't support generics, but it should be fine for now.